server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -6208,8 +6208,8 @@ public VirtualMachine migrateVirtualMachineWithVolume(Long vmId, Host destinatio
             }
         }
 
-        if (!capabilities.isStorageMotionSupported()) {
-            throw new CloudRuntimeException("Migration with storage isn't supported on hypervisor " + srcHost.getHypervisorType() + " of version " + srcHost.getHypervisorVersion());
+        if (capabilities == null || !capabilities.isStorageMotionSupported()) {
```

Comment:
@slavkap when capabilities == null, it might be better to throw relevant message, something like "Cannot migrate VM with storage, as the capabilities are not found for the hypervisor X with version 1".

Reply:
Thanks, @sureshanaparti, for the review! I tried to kill 2 birds with one stone :)
I hope the new changes are better than this one